CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+cmake_minimum_required(VERSION 3.19)
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED on)
```

Comment:
CMake automatically verifies that there is a compiler that supports `c++2a`. It doesn't care about the compiler version, it determines support by running a tiny compilation test that lasts a few ms.

No more users complaining about failed compilation, CMake will tell them that they don't have the right stuff.